test/unit/rand/src/RandTest.cpp

```diff
@@ -95,3 +97,10 @@ TEMPLATE_LIST_TEST_CASE("defaultRandomGeneratorIsWorking", "[rand]", alpaka::tes
 
     REQUIRE(fixture(kernel));
 }
+
+TEMPLATE_LIST_TEST_CASE("defaultRandomGeneratorIsTriviallyCopyable", "[rand]", alpaka::test::TestAccs)
+{
+    using Acc = TestType;
+    using DefaultEngine = decltype(alpaka::rand::engine::createDefault(std::declval<Acc>(), 0u, 0u, 0u));
+    REQUIRE(std::is_trivially_copyable_v<DefaultEngine>);
```

Comment:
Should this not be a `static_assert`? Is it not better to fail as early as possible, i.e. at compile time?

Reply:
I am not sure how we approach it generally. I think runtime is better here, but would keep it uniform to how other tests handle it.

Reply:
Just noticed, that @bernhardmgruber suggested `STATIC_REQUIRE`.

Reply:
Yes, please use `STATIC_REQUIRE` for compile time "unit tests".